quic/s2n-quic-transport/src/transmission/early.rs

```diff
@@ -47,6 +50,21 @@ impl<'a, Config: endpoint::Config> super::Payload for Payload<'a, Config> {
             // inform the ack manager the packet is populated
             self.ack_manager.on_transmit_complete(context);
         }
+
+        // In order to trigger the loss recovery mechanisms during the handshake make all packets
+        // ack-eliciting. This is especially true for the client in order to give the server more
+        // amplification credits.
+        // Only send a PING if:
+        // * We're not congestion limited
+        // * The packet isn't already ack-eliciting
+        // * Another frame was written to the context
+        if !context.transmission_constraint().is_congestion_limited()
+            && !context.ack_elicitation().is_ack_eliciting()
+            && start_capacity != context.remaining_capacity()
+        {
+            // we need to ignore the transmission constraint
+            let _ = context.write_frame(&Ping);
+        }
```

Comment:
Could this move inside the `if` black on lines 39-47, and then maybe you could remove the `!context.transmission_constraint().is_congestion_limited()` check?

Reply:
yeah that works